sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+			  add_to_nogo(Room.current.id, @data) unless Room.current.id == nil
+			  return nil
+			end 
+
+			#main 'perc sigil' DRC.bput and case statement. When sigils are detected they're added to an array
+			case line = DRC.bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about|This is not an appropriate place for that/)
+			when /^You recall.*/
+			  DRC.message('Room searched too recently')
+			  return nil
+			when /You are too distracted/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /Something about|This is not an appropriate place for that/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /^Having recently been searched/
+			  return nil
+			when *@sigil_list
+			  room_sigils.push(line)
+			  DRC.message("Found: #{room_sigils}")
+			end
+		end #while
+
+		#return the array of all sigils discovered in the room (should be 2 unless something went wrong)
+		return room_sigils
+	end
+	
+	#generate the list of all rooms that contain the current room's Room.current.image data
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		#generate the array
+		roomlist = []
+		data.each do |room|
+		  roomlist.push(room.id)
+		end
+
+		#return the full list of rooms
+		return roomlist
+	end #region_room_list
+	
+	#writes a roomid to the nogo list which is subtracted from the walk_list when the script is run in the future
+	def add_to_nogo(banroom, file_data)
+		#initialize a new array if nogo is empty
+		if file_data.SigilInfo['nogo'] == nil
+		  file_data.SigilInfo['nogo'] = []
+		end
+
+		#push the roomid to the array and remove any duplicates
+		file_data.SigilInfo['nogo'].push(banroom)
+		file_data.SigilInfo['nogo'].uniq
+
+		#alert the player which room has been added to the nogo list
+		DRC.message("Unsuitable room for sigil harvesting. Adding Room #{banroom} to no-go list.")
+		File.open("scripts/data/base-sigils.yaml", "w") { |file| file.write(file_data.to_yaml) } #write roomid to file
+		Flags.reset('sigil-danger-room')
+		pause 0.2
+	end #add_to_nogo
+	
+	#reads base hunting data and combines all rooms with rooms from nogo into a single array
+	def generate_nogo_list(room_list)
+		DRC.message("Pruning walk list to avoid combat. Current list size: #{room_list.length}")
+		ban_list = get_data('hunting').hunting_zones.values.flatten.uniq
+		ban_list += @data.SigilInfo['nogo']
+
+		#subtract array of potentially dangerous rooms from the room list that was passed in
+		room_list -= ban_list
+		DRC.message("Walk list has been pruned. New list size: #{room_list.length}")
+		pause 1
+		return room_list #pruned, safer roomlist is returned
+	end #generate_nogo_list
+
+	#remove any rooms in the nogo list from recorded sigil locations before the script exits
+	def prune_final_list
+		ban_list = get_data('hunting').hunting_zones.values.flatten.uniq
```

Comment:
💡 There's a couple places that `get_data('hunting').hunting_zones.values.flatten.uniq` is being made; I suggest moving this to an instance variable and load it once in `initialize` method

Reply:
Moved hunting data to an instance variable